cardstack/cardstack
# CLI - Stabilize app generation dependencies

Closes #799 

## Added
- a command `cardstack pre-build` that runs file generators within a tmpdir, then copies the result into `/blueprints/` on success.
- a blueprints directory containing the output of the generators

## Changed
- cardstack run has fewer responsibilities. It copies the contents of blueprints into a tempdir and yarn installs.
- exec is in its own file, since it's common functionality needed between run and pre-build
- I modified the gitignore so that blueprint js files are checked in. They were ignored because we had made this whole package use TypeScript, and so we ignored all `.js` previously.

## To test

- delete the blueprints directory. Do `cardstack pre-build` to see the generated app recreated.
- modify a file like the README
- `cardstack run` and check that the resulting change shows up in the tmpdir app
